### PR TITLE
JavaClient: fix deprecated proton-j 0.13 api

### PR DESCRIPTION
**Pull request by SreeramGarlapati**

### CODE REVIEW

**Comment by SreeramGarlapati** [comment]
@sbcr is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@hmlam is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@sjkwak is added to the review. #Closed

File: java/pom.xml [hunk 1]
```diff
@@ -43,7 +43,7 @@
 	 	<dependency>
 	 	    <groupId>org.bouncycastle</groupId> 
 	 	    <artifactId>bcpkix-jdk15on</artifactId> 
-	 	    <version>1.54</version> 
+	 	    <version>1.49</version>
```
Comment (SreeramGarlapati) [inline java/pom.xml]: >49 [](start = 18, length = 2)

Had to revert this to 1.4X - as Proton-j 0.13.1 is not ready for 1.54

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/amqp/ConnectionHandler.java [hunk 2]
```diff
@@ -22,27 +28,26 @@
...
+	
 	@Override
 	public void onConnectionInit(Event event)
 	{
-		Connection connection = event.getConnection();
-		connection.setHostname(this.hostname + ":" + ClientConstants.AMQPS_PORT);
-		connection.setContainer(UUID.randomUUID().toString());
+		final Connection connection = event.getConnection();
+		final String hostName = event.getReactor().getConnectionAddress(connection);
+		connection.setHostname(hostName);
```
Comment (SreeramGarlapati) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/amqp/ConnectionHandler.java]: >connection.setHostname(hostName) [](start = 2, length = 32)

Pure AMQP clients wouldn't need this step. This is EventHubs service requirement to add hostname property to Open performative - to infer namespace.

Comment (sjkwak) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/amqp/ConnectionHandler.java]: need to set -- connection.setContainer(UUID.randomUUID().toString());

**Pull request merged**